lib/syskit/roby_app/plugin.rb

```diff
@@ -146,6 +154,50 @@ def self.setup(app)
 
                 rtt_core_model = app.default_loader.task_model_from_name("RTT::TaskContext")
                 Syskit::TaskContext.define_from_orogen(rtt_core_model, register: true)
+
+                # Log Transfer FTP Server spawned during Application#setup
+                @tmp_root_ca = LogTransferIntegration::TmpRootCA.new
+                start_local_log_transfer_server(app.log_dir, @tmp_root_ca)
+                config_log_transfer
+            end
+
+            def self.send_file_transfer_command(name, logfile)
+                # Establishes communication with said process server
+                client = Syskit.conf.process_server_for(name)
+
+                # Commands method log_upload_file from said process server
+                client.log_upload_file(
+                    @log_transfer_ip,
+                    @log_transfer_port,
+                    File.read(@log_transfer_certificate),
+                    @log_transfer_user,
+                    @log_transfer_password,
+                    logfile
+                )
+                client
+            end
+
+            def self.config_log_transfer
+                @log_transfer_port = @log_transfer_server.port
+                @log_transfer_certificate = @tmp_root_ca.certificate
+                @log_transfer_user = @tmp_root_ca.ca_user
+                @log_transfer_password = @tmp_root_ca.ca_password
+                ip = Socket.ip_address_list.detect(&:ipv4_private?)
+                @log_transfer_ip = ip.ip_address
```

Comment:
- What I requested @rbtmrcs does is to make @log_transfer_ip a parameter that **must** be configured per-app. Do not try to guess that IP
- in any case, the purpose of log_transfer_ip is to be the IP from which a **remote** process server can access the FTP server. Picking a private IP is most probably not a very good default.